### PR TITLE
Push docker image to docker.io for release

### PR DESCRIPTION
**Pull request by sebastienvas**

### CODE REVIEW

**Comment by sebastienvas** [comment]
@rvkubiak When moving to Container Builder we need to deal with authenticating to docker.io

File: bin/push-docker [hunk 1]
```diff
@@ -24,13 +24,24 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-hub="gcr.io/istio-testing"
-tags=$(whoami)_$(date +%y%m%d_%h%m%s)
+
+function docker_push() {
+  im="${1}"
+  if [[ "${im}" =~ 'gcr.io' ]]; then
+    gcloud docker -- push "${im}"
```
Comment (kyessenov) [inline bin/push-docker]: this is not necessary. why does gcloud need special docker push?

Comment (sebastienvas) [inline bin/push-docker]: because you need gcloud authentication

Comment (kyessenov) [inline bin/push-docker]: `gcloud docker --authorize-only` is not enough?

**Comment by sebastienvas** [comment]
ahah We ran out of global external ip, and this is why we failed to get an ingress. I still dont understand why this use an global ip, but nevermind.

**Comment by sebastienvas** [comment]
@rshriram Are you referring to the yaml files in istio/istio ? testing artifacts are still going to point to gcr.io/istio-testing, the official releases (starting with the next one) will point to docker.io. Since 0.2.0 is not real release, I don t think there is a point a pushing those tags to docker.io

File: bin/push-docker (hunk 1 above)
Comment (sebastienvas) [inline bin/push-docker]: I removed this. I think those accomplish the same thing. In any case you have to call gcloud for gcr.io images, so what s the issue ?

Comment (kyessenov) [inline bin/push-docker]: One less bash function and one less exception for gcloud :)

**Pull request merged**